staging/src/k8s.io/client-go/kubernetes/clientset.go

```diff
@@ -333,160 +333,163 @@ func (c *Clientset) Discovery() discovery.DiscoveryInterface {
 
 // NewForConfig creates a new Clientset for the given config.
 func NewForConfig(c *rest.Config) (*Clientset, error) {
-	configShallowCopy := *c
-	for _, configCopy := range configShallowCopy.GetAllConfigs() {
+	configShallowCopy := rest.NewAggregatedConfig()
+
+	for _, currentConfig := range c.GetAllConfigs() {
+		configCopy := *currentConfig
 		if configCopy.RateLimiter == nil && configCopy.QPS > 0 {
 			configCopy.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(configCopy.QPS, configCopy.Burst)
 		}
+		configShallowCopy.AddConfig(&configCopy)
 	}
 	var cs Clientset
 	var err error
-	cs.admissionregistrationV1beta1, err = admissionregistrationv1beta1.NewForConfig(&configShallowCopy)
+	cs.admissionregistrationV1beta1, err = admissionregistrationv1beta1.NewForConfig(configShallowCopy)
```

Comment:
I am puzzled by this change. I see the signature admissionregistrationv1beta1.NewForConfig(c *rest.Config) in the code. Could you tell me how come this change works?

Reply:
configShallowCopy has type *rest.Config